domain/secret/service/interface.go

```diff
@@ -56,9 +56,11 @@ type State interface {
 	GetLatestRevision(ctx context.Context, uri *secrets.URI) (int, error)
 	GetLatestRevisions(ctx context.Context, uris []*secrets.URI) (map[string]int, error)
 	GetSecretValue(ctx context.Context, uri *secrets.URI, revision int) (secrets.SecretData, *secrets.ValueRef, error)
-	ListSecrets(ctx context.Context, uri *secrets.URI,
-		revision *int, labels domainsecret.Labels,
-	) ([]*secrets.SecretMetadata, [][]*secrets.SecretRevisionMetadata, error)
+	GetSecretByURI(ctx context.Context, uri secrets.URI, revision *int) (*secrets.SecretMetadata,
+		[]*secrets.SecretRevisionMetadata, error)
+	GetSecretsByLabels(ctx context.Context, labels domainsecret.Labels, revision *int) ([]*secrets.SecretMetadata,
```

Comment:
I guess this should be `ListSecretsByLabels`
When I suggested the api change, I should have suggested List for this one